server/src/main/java/org/opensearch/action/search/SearchShardTask.java

```diff
@@ -50,9 +51,10 @@
  * @opensearch.api
  */
 @PublicApi(since = "1.0.0")
-public class SearchShardTask extends CancellableTask implements SearchBackpressureTask {
+public class SearchShardTask extends CancellableTask implements SearchBackpressureTask, QueryGroupTask {
```

Comment:
Without this, ResourceUsageYTrackerService would not be able to filter out all the query group related tasks. If not for this, we would be assuming every SearchShardTask and SearchTask be part of a query group.

Reply:
We will extend from `QueryGroupTask` instead of `CancellableTask` based on this PR - https://github.com/opensearch-project/OpenSearch/pull/14708. You can include the `QueryGroupTask` class from the PR for now